plugins/modules/proxmox_vm_info.py

```diff
@@ -218,28 +240,26 @@ def main():
     type = module.params["type"]
     vmid = module.params["vmid"]
     name = module.params["name"]
+    config = module.params["config"]
 
     result = dict(changed=False)
 
     if node and proxmox.get_node(node) is None:
         module.fail_json(msg="Node %s doesn't exist in PVE cluster" % node)
 
     vms_cluster_resources = proxmox.get_vms_from_cluster_resources()
-    vms = []
+    cluster_machines = {int(machine["vmid"]): machine for machine in vms_cluster_resources}
+    vms = {}
 
     if type == "lxc":
-        vms = proxmox.get_lxc_vms(vms_cluster_resources, vmid, name, node)
+        vms = proxmox.get_lxc_vms(cluster_machines, vmid, name, node, config)
     elif type == "qemu":
-        vms = proxmox.get_qemu_vms(vms_cluster_resources, vmid, name, node)
+        vms = proxmox.get_qemu_vms(cluster_machines, vmid, name, node, config)
     else:
-        vms = proxmox.get_qemu_vms(
-            vms_cluster_resources,
-            vmid,
-            name,
-            node,
-        ) + proxmox.get_lxc_vms(vms_cluster_resources, vmid, name, node)
-
-    result["proxmox_vms"] = vms
+        vms = proxmox.get_qemu_vms(cluster_machines, vmid, name, node, config)
+        vms.update(proxmox.get_lxc_vms(cluster_machines, vmid, name, node, config))
+
+    result["proxmox_vms"] = list(vms.values())
```

Comment:
Please note that the unit tests are still failing. My guess is that this is due to the order of the VMs returned. How about sorting them by vmid:
```suggestion
    result["proxmox_vms"] = [vm for vmid, vm in sorted(vms.items())]
```